03week/towersOfHanoi.js

```diff
@@ -19,24 +19,117 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+/*Code Plan
+1. Write checkForWin function to check if all of the numbers are in the b row in order
+    *I started with stacks.b.toString() = "4,3,2,1"
+     but then I wrote the reset function to add another level
+    *If stacks.a and stacks.c are empty (and the isLegal function works)
+     then the all of the numbers used must be in order in stacks.b
+2. Write isLegal function to check:
+    1. Did the user enter a, b, or c?
+    2. Did the user pick from a row that has a number?
+    3. Did the user choose a smaller number to go on a bigger number?
+3. Write movePiece function to append the last number in startStack to endStack
+    *Push the pop of start stack to the end stack
+4. Figure out what ACA means by reset function and write it
+Bonus. Add more levels!!!
+*/
+
+//detaches the last number in the array of the start stack
+//and appends it to the end stack AT THE SAME TIME
+function movePiece(startStack, endStack) {
+  stacks[endStack].push(stacks[startStack].pop());
+}
 
+//resets the stack to the starting position
+function resetGame() {
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  };
 }
 
-function isLegal() {
-  // Your code here
+//The isLegal function first checks if the player wanted to reset the game, if not
+//then it makes sure the player chose actual stacks, that have something in them
+//then it makes sure the player puts the chosen number on top of a number that isn't itself or smaller
+function isLegal(startStack, endStack) {
+  if(startStack === "reset" || endStack === "reset"){
+    resetGame();
```

Comment:
resetGame shouldn't be called in isLegal, it should be called in the parent function. Your isLegal function should only be concerned with checking if the move is legal